### ISSUE TITLE
raw2outputs pathloss calculation

### ISSUE THREAD

**Issue by antifakerr** [XPengZhao/NeRF2#1]
Dear author,

Thanks for sharing the code! Amazing project.

I have trouble understanding the raw2output function in the renderer.py file. I am confused about the way pathloss is calculated:
```
        alpha = raw2alpha(att_a, dists)  # [N_rays, N_samples]
        phase = raw2phase(att_p, dists)

        att_i = torch.cumprod(torch.cat([torch.ones((alpha.shape[0], 1)).cuda(), 1.-alpha + 1e-10], -1), -1)[:, :-1]
        path = torch.cat([r_vals[...,1:], torch.Tensor([1e10]).cuda().expand(r_vals[...,:1].shape)], -1)
        path_loss = 0.025 / path     #<-- this line does not make sense to me

        phase_i = torch.cumsum(torch.cat([torch.ones((alpha.shape[0], 1)).cuda(), phase], -1), -1)[:, :-1]
        phase_i = torch.exp(1j*phase_i)    # [N_rays, N_samples]


        receive_signal = torch.sum(s_a*torch.exp(1j*s_p)*att_i*phase_i*path_loss, -1)  # [N_rays]
        receive_signal = abs(receive_signal)
```
Could you please explain why `path_loss` is calculated this way? **What's the physical model that supports this equation?** Also, it seems to me that the alpha, calculated from the `att_a`, which is the prediction output of the attenuation network is not really used in the actual attenuation calculation. Please correct me if I am wrong.

Thank you in advance for clarification!

**Comment by XPengZhao** [comment]
## Issue 1

Thanks for your question. We will clarify this in our code. The path_loss calculation in the code is based on the free-space path loss (FSPL) model, which measures the power density (attenuation) reduction of an electromagnetic wave as it propagates through free space. In the FSPL model, the power received at a distance d from the transmitter is given by:

$P_r = \left(\frac{\lambda}{4 \pi d} \right)^2 P_t $

Here, $\lambda$ is the wavelength of the transmitted signal, and $P_t$ is the transmitted power. Since the NeRF<sup>2</sup> system is predicting the amplitude of an RFID signal operating at 920 MHz, we can calculate the associated wavelength and substitute it into the FSPL equation to derive the following form that is used in the code:

$U_r = \frac{\lambda}{4 \pi d}  U_t  \approx \frac{0.025}{d} U_t$



## Issue 2

Regarding the use of `att_a` and `alpha`, the `att_i` variable in the code is derived from `alpha`, which is in turn computed from `att_a` as you mentioned. This is evident in this line of the code:

```
att_i = torch.cumprod(torch.cat([torch.ones((alpha.shape[0], 1)).cuda(), 1.-alpha + 1e-10], -1), -1)[:, :-1]
```

Additionally, `att_i` is indeed used in the attenuation calculation, as seen in this line of the code:

```
receive_signal = torch.sum(s_a*torch.exp(1j*s_p)*att_i*phase_i*path_loss, -1)  # [N_rays]
```

This implementation aims to guide the NeRF<sup>2</sup> system to learn material-dependent attenuation, while the path-dependent attenuation is manually introduced through the FSPL calculation. This approach allows the NeRF<sup>2</sup> system to focus on understanding the material properties that influence signal attenuation, rather than the path-dependent factors.
